instructions/04-connecting-to-confluent-cloud.md

```diff
@@ -0,0 +1,218 @@
+
+**WARNING:** Do not use a production environment for this course. Use a new environment with new credentials. You don't want to accidentally compromise your production environment.
+
+![Add Cloud Environment](images/04-connecting-to-confluent-cloud/add-cloud-environment.png)
+
+- From the **Environments** menu click **+ Add cloud environment**.
+- Name your environment `flink-table-api-java`.
+- Use the **Essentials** Stream Governance package.
+- You will be asked to create a cluster. Choose **I'll do it later**.
+
+### Command Line Reference
+
+Create the environment.
+
+```
+confluent environment create flink-table-api-java
+```
+
+Set the active environment.
+
+```
+confluent environment use <environment id>
+```
+
+Enable the schema registry (adjust cloud and geo settings as required).
+
+```
+confluent schema-registry cluster enable --cloud gcp --geo us --package essentials
+```
+
+## Create a compute pool.
+
+Flink uses compute resources and requires a compute pool. Let's create one.
+
+![Add Compute Pool](images/04-connecting-to-confluent-cloud/add-compute-pool.png)
+
+- Inside the `flink-table-api-java` environment, select **Flink** -> **Compute Pools** and click **Create compute pool**.
+- Select an appropriate cloud and region.
+- Give your pool the name `marketplace-compute`.
+
+### Command Line Reference
+
+Create a compute pool (adjust cloud and region settings as required).
+
+```
+confluent flink compute-pool create marketplace-compute --cloud gcp --region us-central1 --max-cfu 10
+```
+
+## Create an Apache Flink API key.
+
+To access Flink in Confluent Cloud, we need an API key.
+
+![Add API Key](images/04-connecting-to-confluent-cloud/add-api-key.png)
+
+- Still in the **Flink** tab, select **API Keys** and click **Add API key**.
+- Use the **My Access** key type.
+	- **NOTE:** **Granular Access** is recommended for production.
+- Add an appropriate name and description (eg. `marketplace-api-key`) for the key.
+- Download and save the key somewhere that you can access it later.
+
+### Command Line Reference
+
+Create a Flink API Key (adjust cloud and region settings as required).
+
+**NOTE:** Save the key and secret for later.
+
+```
+confluent api-key create --resource flink --cloud gcp --region us-central1
+```
+
+## Download the Table API plugin.
+
+**NOTE:** This step is temporary. Once the plugin has been released this will not be required.
+
+- Clone the `confluent-flink-plugin` from [Github](https://github.com/confluentinc/confluent-flink-plugin-private)
+- Run the following command from the repository:
+
+  ```
+  mvn clean install
+  ```
+
+## Download the code.
+
+Now that we have an environment, we need an application to connect to it. If you haven't already done so clone the Github repo:
+
+[Clone it on Gitub](https://github.com/confluentinc/learn-apache-flink-table-api-for-java-exercises)
+
+You will need a suitable Java development environment including:
+
+- Java 17
```

Comment:
I assume Java 17 is a Flink requirement/dependency. What if someone has a more recent version? Will things fail? Is it still true that you can have multiple versions installed, as long as you have your JAVA_HOME set properly?

Reply:
Actually, we are targeting Java 21 for this. I will update it.